PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/OriginalGraph.java

```diff
@@ -6,17 +6,23 @@
 public class OriginalGraph implements GraphInterface {
 
   private HashMap<Integer, Node> nodes;
+  private HashMap<Integer, AbstractNode> abstractNodes;
```

Comment:
Why did you add this variable? It seems like an exact copy of variable `nodes`, but with different generics (since you call `put()` on both of them each time you add anything to either of them).
If this variable is really only needed for `getAbstractNodes()` and contains the same objects as `nodes`, then I suggest you change the `getAbstractNodes()` to:
``` java
  @Override
  public HashMap<Integer, ? extends AbstractNode> getAbstractNodes() {
    return nodes;
  }
```
And don't forget to change the method in the interface to:
``` java
HashMap<Integer, ? extends AbstractNode> getAbstractNodes();
```